### DISCUSSION TITLE
Mujoco renderer error

### DISCUSSION THREAD

**Discussion by Aristomo** [google-deepmind/mujoco#685] [Asking for Help]
I just run the sample code in the tutorial of python bindings,

import mujoco
xml = """
<mujoco>
  <worldbody>
    <light name="top" pos="0 0 1"/>
    <geom name="red_box" type="box" size=".2 .2 .2" rgba="1 0 0 1"/>
    <geom name="green_sphere" pos=".2 .2 .2" size=".1" rgba="0 1 0 1"/>
  </worldbody>
</mujoco>
"""
model1 = mujoco.MjModel.from_xml_string(xml)
data = mujoco.MjData(model1)
renderer = mujoco.renderer.Renderer(model=model1)

Then there is an error:

Exception ignored in: <function GLContext.__del__ at 0x000002BE7D526670>
Traceback (most recent call last):
  File "D:\anaconda\lib\site-packages\mujoco\glfw\__init__.py", line 41, in __del__
  File "D:\anaconda\lib\site-packages\mujoco\glfw\__init__.py", line 35, in free
TypeError: 'NoneType' object is not callable

Maybe because I did not initialize the render? How to solve it?

**Comment by Amritpal-001** [comment]
Did anyone find any fix for this?

**Reply by Aristomo** [reply]
You can remove the "()" behind the "free" function in "glfw_init_.py" and the position can be located if you click the error information.

**Comment by mnm117** [comment]
Hello @Amritpal-001, @Aristomo 
to solve this problem, you can install Mujoco 3.1.1 and then use one of the below options:

`
# Option 1.
renderer = Renderer(...)
// use renderer
renderer.close()

# Option 2.
with Renderer(...) as renderer:
// use renderer`

for example :
renderer = mujoco.Renderer(model)
renderer.close()


this works for me
